Source/JavaScriptCore/runtime/JSMicrotask.cpp

```diff
@@ -168,6 +172,58 @@ static void promiseResolveThenableJob(JSGlobalObject* globalObject, JSValue prom
     EXCEPTION_ASSERT(scope.exception() || true);
 }
 
+static void asyncFromSyncIteratorContinueOrDone(JSGlobalObject* globalObject, VM& vm, JSValue context, JSValue result, JSPromise::Status status, bool done)
+{
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    auto* contextObject = asObject(context);
+    JSValue promise = contextObject->getDirect(vm, vm.propertyNames->builtinNames().promisePrivateName());
+    ASSERT(promise.inherits<JSPromise>());
+
+    switch (status) {
+    case JSPromise::Status::Pending: {
+        RELEASE_ASSERT_NOT_REACHED();
+        break;
+    }
+    case JSPromise::Status::Rejected: {
+        JSValue syncIterator = contextObject->getDirect(vm, vm.propertyNames->builtinNames().syncIteratorPrivateName());
+        if (syncIterator.isObject()) {
+            JSValue returnMethod;
+            JSValue error;
+            {
+                auto catchScope = DECLARE_CATCH_SCOPE(vm);
+                returnMethod = asObject(syncIterator)->get(globalObject, vm.propertyNames->returnKeyword);
+                if (catchScope.exception()) [[unlikely]] {
+                    error = catchScope.exception()->value();
+                    if (!catchScope.clearExceptionExceptTermination()) [[unlikely]] {
+                        scope.release();
+                        return;
+                    }
+                }
+            }
+            if (error) [[unlikely]] {
+                jsCast<JSPromise*>(promise)->reject(vm, globalObject, error);
+                return;
+            }
+            if (returnMethod.isCallable()) {
+                callMicrotask(globalObject, returnMethod, syncIterator, dynamicCastToCell(returnMethod), ArgList { }, "returnMethod is not a function"_s);
```

Comment:
`returnMethod` is just our variable name, so I think it should not be used for error message
```suggestion
                callMicrotask(globalObject, returnMethod, syncIterator, dynamicCastToCell(returnMethod), ArgList { }, "return is not a function"_s);
```

Reply:
fixed.